src/forms/formHelpers.ts

```diff
@@ -17,17 +17,24 @@ export const zStringISODateUTC = z
 
 export const zEthAddress = z
   .string()
+  // `resolveName` throws outright if it is passed a malformed eth address
+  // so we need to refine it first so safeParseAsync doesn't throw
+  // unexpectedly
+  .refine(validAddressOrENS, 'Wallet address is invalid')
   .transform(s => provider.resolveName(s))
   .transform(s => s || '')
-  .refine(s => ethers.utils.isAddress(s), 'Wallet address is invalid')
+  .refine(s => s, 'unresolved ENS name')
```

Comment:
the `s => s | ''` insures this never happens, I think these don't make sense now..
Maybe this one we call `zEthAddressEns` and the other `zEthAddress`